lms/services/vitalsource.py

```diff
@@ -0,0 +1,60 @@
+import oauthlib
+from oauthlib.oauth1 import SIGNATURE_HMAC_SHA1, SIGNATURE_TYPE_BODY
+
+
+class VitalSourceService:
+    def __init__(self, _context, request):
+        self._oauth_key = request.registry.settings["vitalsource_launch_key"]
+        self._oauth_secret = request.registry.settings["vitalsource_launch_secret"]
```

Comment:
This unfortunately crashes with `ValueError: Only unicode objects are unescapable` on launching a VitalSource assignment if these settings are missing from the environment. I wonder if we could at least turn it into a better traceback? Maybe even just `assert self._oauth_key is not None and self._oauth_secret is not None` here

Reply:
Could we rename `oauth_key` and `oauth_secret` so that 1. The naming is consistent with the names of the settings (`launch_key` and `launch_secret`); 2. The names make it clear that these are a key and secret for doing LTI launches.

They're called `oauth_consumer_key` and `oauth_consumer_secret` in the LTI 1.1 spec.

I think maybe `vitalsource_lti_launch_key` and `vitalsource_lti_launch_secret` would be good names for the settings, and then just `_lti_launch_key` and `_lti_launch_secret` for the variables within `VitalSourceService`

Reply:
> his unfortunately crashes with ValueError: Only unicode objects are unescapable on launching a VitalSource assignment if these settings are missing from the environment. I wonder if we could at least turn it into a better traceback?

`VitalSourceService.__init__` now raises an error if these settings are missing. It feels like we could do with a better pattern for handling registry keys that are valid names but not set. The current way that the registry is populated results in a `None` entries for most key names which are valid but for which no value is set.

> I think maybe vitalsource_lti_launch_key and vitalsource_lti_launch_secret would be good names for the settings, and then just _lti_launch_key and _lti_launch_secret for the variables within VitalSourceService

I agree. I've updated the environment variables, settings and the params formerly known as oauth_{key, secret} accordingly.